### DISCUSSION TITLE
User Attempt History

### DISCUSSION THREAD

**Discussion by WillieCubed** [acmutd/comet_planning#46] [Ideas]
A student must be able to input and view information about the courses they've taken.

This information includes:
- What course was taken
- When a course was taken

Is there any information should be included here?

**Comment by wbne** [comment]
I think the grade the User has received should be included (assuming they have finished the course).
The grade would be a letter grade including the modifiers so A+, A, A-, etc. would count. 
Also W, I, NC, and CR should be included

**Reply by wbne** [reply]
Another piece of data should be the professor name but this is less important. Having the professor name would simply allow us to connect their reviews of courses to professors and vice versa. While it isn't necessary, I think it would allow for more interesting data correlations.

**Reply by wbne** [reply]
Also the date format the course was taken should probably include the year taken and what semester. Any course over the summer is just Summer including 5w and 8w courses.

**Reply by WillieCubed** [reply]
Some notes:
- All of the possible grades are and their GPAs are currently stored [here](https://github.com/acmutd/nebula-web/blob/d3df748a616fd37523c7fafd88cd1b05fed39066/modules/common/data.ts) and follow the same GPA weighting as defined in the Catalog.
- Part of the purpose of allowing a user to record their course history is allowing us to generate an _unofficial_ unofficial transcript in addition to validating what courses one has remaining. We should add instructor information to a course attempt, as well.
  - Furthermore, we can use historical course data to allow match to existing courses instead of manually inputting instructor names.
  - Through the Nebula Data Service, we're going to be recording unique instructors, so we should be able to create class section objects that refer to those instructors.
- As for recording when a course was taken, we'll stick to the [`SemesterCode`](https://github.com/acmutd/nebula-web/blob/d3df748a616fd37523c7fafd88cd1b05fed39066/modules/common/data.ts#L36) identifiers in Nebula Web (in our case, just fall (f), spring (s), and summer (u).
